feat(nix/{holonix,holochain}): use makeOverridable and demo with chc feature

steveej opened the pull request:
this allows overriding any argument passed to `craneLib.buildPackage`. the new `holochain_chc` package demonstrates how to use it for passing additional arguments to cargo.

### Summary

cc @zo-el FYI


### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

ThetaSinner commented on nix/modules/holochain.nix:
```diff
@@ -55,14 +55,16 @@
         passthru.src.rev = flake.config.reconciledInputs.holochain.rev;
       });
 
+      holochain_chc = holochain.override { cargoExtraArgs = " --features chc"; };
```
Does this mean we'll always build `holochain `+ `chc` on CI as a different build of the `holochain` binary? If so I'd prefer this to be a doc example.

I'm a fan of testing features in general but I think it's too slow on PR runs

jost-s replied:
Right, and if that's so, what about cachix?

steveej replied:
we don't build this on CI automatically. 

it's exposed for whoever wants it and it'll require compiling.

ThetaSinner reviewed (approved):
Nice! This also enables people turning on sqlit encryption when building with Nix which is a much needed ability.

zo-el commented:
Can we add some documentation for the feature this enables?

steveej commented:
> Can we add some documentation for the feature this enables?

yes, please help me understand what level of documentation would be helpful. here's a suggestion for me to gauge what would be helpful.

---

feat(flake/packages/holochain): make customisable via [lib.makeOverridable](https://nixos.org/manual/nixpkgs/stable/#sec-lib-makeOverridable) 

this means that you can pass e.g. `holochain.override { cargoExtraArgs = "--feature chc"; }` or any other desirable attribute to override the attributes that are passed to [craneLib.buildPackage](https://crane.dev/API.html#cranelibbuildpackage).

steveej commented:
i added the `override` to holonix as well and added an example of how this can be used by end-users. 

i've not added documentation that shows specifically how to just do `holochain.override ...` because it's not part of the holonix API. 

any suggestions where to add that example?

steveej commented:
also i'm not sure this is good UX. i'm virtually demoting this PR to an RFC. i'm happy to go over other ways of customizing flake outputs (this is not part of the flake standard yet so we have no clear guides).

jost-s commented on holonix/README.md:
```diff
@@ -45,3 +45,14 @@ inputs = {
+
+This means that you can pass e.g. `holochain.override { cargoExtraArgs = " --feature chc"; }` or any other desirable attribute to override the attributes that are passed to [craneLib.buildPackage](https://crane.dev/API.html#cranelibbuildpackage).
+
+Creating a devShell based on holonix this, can be achieved by passing `holonix.override { holochainOverrides = { cargoExtraArgs = "--features chc"; }}`.
```
This sentence is funny

jost-s reviewed (commented):
I think the example is in a good location. The UX seems fine to me, a bit verbose but I don't know if these customizations come in shorthand form.

steveej commented on holonix/README.md:
```diff
@@ -45,3 +45,14 @@ inputs = {
+
+This means that you can pass e.g. `holochain.override { cargoExtraArgs = " --feature chc"; }` or any other desirable attribute to override the attributes that are passed to [craneLib.buildPackage](https://crane.dev/API.html#cranelibbuildpackage).
+
+Creating a devShell based on holonix this, can be achieved by passing `holonix.override { holochainOverrides = { cargoExtraArgs = "--features chc"; }}`.
```
well i wish i could retain the funnyness and make it understandable too :laughing: for now i'll focus on the latter. how about this?

```suggestion
Creating a devShell based on holonix, passing on overrides to the holochain package can be achieved by passing for example `holonix.override { holochainOverrides = { cargoExtraArgs = "--features chc"; }}`.
```

jost-s replied:
Ah, now I understand what you wanted to say. I think the latter was almost understandable then =) I suggest
```suggestion
In a devShell based on holonix, this can be achieved by defining `holonix.override { holochainOverrides = { cargoExtraArgs = "--features chc"; }}`.
```
